extension/src/popup/components/signTransaction/TransactionInfo/index.tsx

```diff
@@ -3,18 +3,26 @@ import { Icon, IconButton } from "@stellar/design-system";
 import { useTranslation } from "react-i18next";
 
 import { stroopToXlm } from "helpers/stellar";
+import { MEMO_TYPES } from "popup/constants/memoTypes";
 
 import "./styles.scss";
 
 const MemoDisplay = ({
   memo,
   isMemoRequired,
 }: {
-  memo: string;
+  memo: { value: string; type: MEMO_TYPES };
```

Comment:
Should this type be a `Memo` from stellar-base?
https://github.com/stellar/js-stellar-base/blob/cf370f4e627e2b0e296ceceaebd0164dede396bd/types/index.d.ts#L212

Reply:
Good call - I think we should leverage types from the SDK wherever we can

Reply:
Here we need to keep `memo: { value: string; type: MemoType };` because this is for decoded memo.